src/controllers/TerminalController.java

```diff
@@ -75,6 +78,53 @@ private void exit() {
         context.exit();
     }
 
+    private String processSpelling(String inputtedCommand) {
+        int maxDistance = (int) Math.ceil(inputtedCommand.length() / 8f);
+        LevenshteinDistance levenshteinDistance = new LevenshteinDistance();
+        int currentMin = Integer.MAX_VALUE;
+        String newCommand = inputtedCommand;
+        for (String command : AllCommands().keySet()) {
+            int dist = levenshteinDistance.apply(command, inputtedCommand);
+            if (dist <= maxDistance) {
+                if (dist < currentMin){
+                    newCommand = command;
+                    currentMin = dist;
+                }
+            }
+        }
+        return newCommand;
+    }
+
+    private Integer processNumber(String inputtedCommand) {
+        if (NumberUtils.isParsable(inputtedCommand)){
+            int number = NumberUtils.createInteger(inputtedCommand) - 1;
+            if (number <= AllCommands().size() - 1 & number >= 0) {
+                return number;
+            }
+        }
+        return null;
+    }
+
+    private String getCommand(String inputtedCommand) {
+        if (AllCommands().containsKey(inputtedCommand)) {
+            return inputtedCommand;
+        }
+
+        Integer numberCommand = processNumber(inputtedCommand);
+        if (numberCommand != null) {
+            return new ArrayList<>(AllCommands().keySet()).get(numberCommand);
+        }
+
+        String correctSpelling = processSpelling(inputtedCommand);
+        if (!Objects.equals(correctSpelling, inputtedCommand)) {
```

Comment:
The if statement is un-necessary. Due to the first if statement in the method, if the inputted command has the correct spelling we just return the command associated with the input.

Reply:
Done!